### ISSUE TITLE
Add PDF guide

### ISSUE THREAD

**Issue by mxschmitt** [microsoft/playwright#7822]
Customers are confused, when a PDF results in a PDF viewer and when in a download event. We should explain how to workaround it in the relevant browsers.

**Comment by corradin** [comment]
@mxschmitt Did you also take a look at the solution I proposed in case people are not able to change the source code of the application under test? https://github.com/microsoft/playwright/issues/6342#issuecomment-929112605

This uses the download event for headless chrome and the page event for all other browsers that use the integrated PDF reader.

**Comment by mxschmitt** [comment]
@corradin yes I've seen that. Creative solution! Question from our side, what in your opinion gets mostly asserted / tested when working with PDF files?

waitForResponse() seems working in all the cases but it seems important to assert the suggested file name?

**Comment by jvarghese2022** [comment]
Is there any simpler way to handle to pdf viewer where the url starts as blob.

**Comment by mandras73** [comment]
Do you have a python version of the workaround?

**Comment by vincenzo-gasparo** [comment]
Here's a python version of the workaround:

```python
import re
from playwright.sync_api import Page, expect


def test_download_pdf(page: Page):
    page.goto("https://www.example.com")

    def handle_pdf(route):
        response = page.context.request.get(route.request)
        route.fulfill(
            response,
            headers={**response.headers, "Content-Disposition": "attachment"},
        )

    page.route(re.compile(r".*\.pdf"), lambda route: handle_pdf(route))

    with page.expect_download() as download_info:
        page.get_by_text("click_here_to_download_pdf").click()
```

**Comment by kinopop** [comment]
> Here's a python version of the workaround:
> 
> ```python
> import re
> from playwright.sync_api import Page, expect
> 
> 
> def test_download_pdf(page: Page):
>     page.goto("https://www.example.com")
> 
>     def handle_pdf(route):
>         response = page.context.request.get(route.request)
>         route.fulfill(
>             response,
>             headers={**response.headers, "Content-Disposition": "attachment"},
>         )
> 
>     page.route(re.compile(r".*\.pdf"), lambda route: handle_pdf(route))
> 
>     with page.expect_download() as download_info:
>         page.get_by_text("click_here_to_download_pdf").click()
> ```

What plug-ins do I lack?
>             response,
error 1 argument
>             response=response,
is that true？
my error ：waiting for event " download"

**Comment by cobaltautomationdev** [comment]
@beegotsy  
I have implemented a feature in my code that triggers a Chrome PDF preview popup when a specific button labeled "PO Print for Vendor" is clicked. Upon inspecting the elements within this popup, I found that there is limited interaction available; the primary options are to save the document as a PDF or print it.

```
async with page.expect_popup() as popup_info:
            await page.locator("#quick-process").get_by_text("PO Print for Vendor").click()
popup_page = await popup_info.value
```
To automate the process of saving the PDF, I attempted to invoke the print function using evaluate on the popup page object, like so:
`await popup_page.evaluate('''() => {window.print();}''')`

This approach successfully triggers the print dialog, which includes an option to save the document as a PDF. However, this method requires manual intervention to input a filename and confirm the save action, which prevents the automation from completing without user input.
